src/sunstone/OpenNebulaAddons.rb

```diff
@@ -0,0 +1,89 @@
+# -------------------------------------------------------------------------- #
+# Copyright 2002-2018, OpenNebula Project, OpenNebula Systems                #
+#                                                                            #
+# Licensed under the Apache License, Version 2.0 (the "License"); you may    #
+# not use this file except in compliance with the License. You may obtain    #
+# a copy of the License at                                                   #
+#                                                                            #
+# http://www.apache.org/licenses/LICENSE-2.0                                 #
+#                                                                            #
+# Unless required by applicable law or agreed to in writing, software        #
+# distributed under the License is distributed on an "AS IS" BASIS,          #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   #
+# See the License for the specific language governing permissions and        #
+# limitations under the License.                                             #
+#--------------------------------------------------------------------------- #
+
+require 'rubygems'
+require 'json'
+
+ADDONS_LOCATION = SUNSTONE_LOCATION + "/public/app/addons/dist/*"
+MAIN_DIST_PATH = SUNSTONE_LOCATION + "/public/dist/main-dist.js"
+MAIN_PATH = VAR_LOCATION + "/sunstone/main.js"
+
+class OpenNebulaAddons
+    def initialize(logger)
+        @logger = logger
+        @main_folder = File.join(VAR_LOCATION, "sunstone")
+
+        begin
+            Dir.mkdir(@main_folder)
+        rescue Exception => e
+            @logger.error "Cannot create sunstone folder"
+            @logger.error e.message
+        end
+
+        main = File.open(MAIN_PATH, "w")
```

Comment:
Should be new? also is better to set permissions for first time: File::CREAT|File::TRUNC|File::RDWR, 0644